gap/digraph.gi

```diff
@@ -591,19 +591,33 @@ function(D)
 end);
 
 InstallMethod(String, "for an immutable digraph by out-neighbours",
```

Comment:
I like what you've done. But I think it should further integrated so that it is used as the result for `PrintObj`and `PrintString` too. (See [this part of the GAP manual](https://www.gap-system.org/Manuals/doc/ref/chap6.html#X8074A8387C9DB9A8) for more information about this topic).

With this PR as it is, you get the following:
```
gap> gr := RandomDigraph(6);;
gap> Print(gr);
Digraph( IsImmutableDigraph, [ [ ], [ 2, 4 ], [ 3, 4 ], [ 1, 2, 6 ], [ 2, 4, \
5 ], [ 3, 5, 6 ] ] )
gap> PrintString(gr);
"Digraph( IsImmutableDigraph, [ [ ], [ 2, 4 ], [ 3, 4 ], [ 1, 2, 6 ], [ 2, 4,\
 5 ], [ 3, 5, 6 ] ] )"
gap> String(gr);
"DigraphFromDigraph6String(\"&E?SKpUJ\");"
```
But I think `Print` should delegate to `PrintString` which should delegate to `String` (in the appropriate way), so that we get the short thing in basically all cases 🙂

Reply:
Nice idea - I've implemented that now. Do you think I should rename the documentation as being for `Print`, rather than for `String`? I think this would make sense, since now `Print(D)` achieves the result of `String(D); Print(last);`.

Reply:
I don’t mind. You could have the documentation refer to both print and string, in the way that DigraphTransitiveClosure and DigraphReflexiveTransitiveClosire are documented together.